src/casp/services/api/data_manager/cell_operations.py

```diff
@@ -17,15 +19,20 @@ class CellOperationsDataManager(BaseDataManager):
     """
 
     # Directories for SQL templates
-    CELL_ANALYSIS_TEMPLATE_DIR = f"{settings.SERVICES_DIR}/api/data_manager/sql_templates/cell_operations"
+    CELL_OPERATIONS_TEMPLATE_DIR = f"{settings.SERVICES_DIR}/api/data_manager/sql_templates/cell_operations"
 
     # SQL template file paths
-    SQL_MATCH_METADATA = f"{CELL_ANALYSIS_TEMPLATE_DIR}/get_neighborhood_distance_summary.sql.mako"
+    SQL_MATCH_METADATA = f"{CELL_OPERATIONS_TEMPLATE_DIR}/get_neighborhood_distance_summary.sql.mako"
     SQL_MATCH_METADATA_DEV_DETAILS = (
-        f"{CELL_ANALYSIS_TEMPLATE_DIR}/get_neighborhood_distance_summary_dev_details.sql.mako"
+        f"{CELL_OPERATIONS_TEMPLATE_DIR}/get_neighborhood_distance_summary_dev_details.sql.mako"
     )
-    SQL_GET_CELLS_BY_IDS = f"{CELL_ANALYSIS_TEMPLATE_DIR}/get_cell_metadata_by_ids.sql.mako"
+    SQL_GET_CELLS_BY_IDS = f"{CELL_OPERATIONS_TEMPLATE_DIR}/create_temp_table_for_metadata.sql.mako.mako"
+    # SQLAlchemy parsable templates
+    SQL_CREATE_CELL_TEMP_TABLE = f"{CELL_OPERATIONS_TEMPLATE_DIR}/create_temp_table_for_metadata.sql"
+    SQL_GET_CELL_METADATA_FROM_TEMP_TABLE = f"{CELL_OPERATIONS_TEMPLATE_DIR}/get_cell_metadata_from_temp_table.sql"
     redis_cache_key_prefix = "cell_operations"
+    # Settings and constants
+    MAX_CELL_IDS_PER_QUERY = 20_000  # Maximum number of cell IDs that can be queried at once
```

Comment:
should this be configurable?

Reply:
Do you mean we need to factor this out to settings?

Reply:
Yeah exactly.